src/box/txn_limbo.c

```diff
@@ -72,6 +72,21 @@ txn_limbo_is_ro(struct txn_limbo *limbo)
 	       (limbo->owner_id != instance_id || txn_limbo_is_frozen(limbo));
 }
 
+void
+txn_limbo_complete(struct txn *txn, bool is_success)
+{
+	/*
+	 * Some rollback/commit triggers require the in_txn fiber
+	 * variable to be set.
+	 */
+	fiber_set_txn(fiber(), txn);
```

Comment:
While you're at it, let's please fix the effective user/session for sync space on_commit trigger as well (like it was done in #7010 for async spaces).

Otherwise I get this with a sync space and 2 nodes (`replication_synchro_quorum= 2`):
```
tarantool> print(box.session.effective_user())
guest
---
...

tarantool> box.begin() box.space.sync:replace{1} box.on_commit(function() print(box.session.effective_user()) end) box.commit()
admin
---
...

tarantool> print(box.session.effective_user())
guest
---
...


```

Reply:
Ok. Fixed